src/quantuminspire/qiskit/qi_job.py

```diff
@@ -66,12 +77,14 @@ def submit(self) -> None:
             raise JobError('Job has already been submitted!')
         self._job_id = self._backend.run(self._qobj)
 
-    def result(self, timeout: Optional[float] = None, wait: float = 0.5) -> QIResult:
+    def result(self, timeout: Optional[float] = None, wait: float = 0.5, only_latest_run: bool = True) -> QIResult:
```

Comment:
You don't test for the different values of `only_latest_run`. Only the default value is tested.

Reply:
I thought about that but the test would actually be no different than: TestQIJob.test_result() currently is because the parameter is redirected to get_experiment_results. I would mock get_experiment_results in that test the same as done in this test, with multiple experiment_results in the return_value. The get_experiment_results is tested somewhere else for the parameter being False and True.